yt/frontends/enzo_p/data_structures.py

```diff
@@ -62,10 +62,19 @@ def __init__(self, id, index, block_name, filename = None):
         self._children_ids = None
         self._parent_id = -1
         self.Level = -1
+        self.particle_count = None
 
     def __repr__(self):
         return "EnzoPGrid_%04d" % self.id
 
+    def _prepare_grid(self):
+        """ Copies all the appropriate attributes from the index. """
+        h = self.index # cache it
```

Comment:
If you follow PEP 8, you want an extra whitespace before the inline `#`.

Reply:
This and below were just copied from another frontend, but I can change them.  Thank you for your comments.

Reply:
You're welcome, @brittonsmith. I understand that other (older) frontends are not pep8-compliant. I sparked that discussion with @ngoldbaum earlier; for reference: https://github.com/yt-project/yt/pull/1506#pullrequestreview-52248147